backend/dataall/modules/dataset_sharing/services/share_processors/s3_access_point_process_share.py

```diff
@@ -204,18 +215,68 @@ def clean_up_share(
             source_env_group,
             env_group,
         )
-        clean_up = clean_up_folder.delete_access_point(
-            share=share,
-            dataset=dataset
-        )
+        clean_up = clean_up_folder.delete_access_point(share=share, dataset=dataset)
 
         if clean_up:
             clean_up_folder.delete_target_role_access_policy(
-                share=share,
-                dataset=dataset,
-                target_environment=target_environment
+                share=share, dataset=dataset, target_environment=target_environment
             )
             if not dataset.imported or dataset.importedKmsKey:
                 clean_up_folder.delete_dataset_bucket_key_policy(dataset=dataset)
 
         return True
+
```

Comment:
Suggestion, we could add the `verify_shares` method as an abstract class method for the managers to force that the method gets implemented

Reply:
will do !